src/dq_suite/common.py

```diff
@@ -54,7 +55,54 @@ def __getitem__(self, key) -> str | Dict[str, Any] | int | None:
         raise KeyError(key)
 
 
-RulesList = list[Rule]  # a list of DQ rules
+@dataclass
+class GeoRule(Rule):
+    """A specialized rule for geospatial validation."""
+    rule_type: Literal["geo"] = "geo"
+    geo_query_template: str | None = None
+    description: str | None = None
+
+    def __post_init__(self):
+        super().__post_init__()
+        if self.rule_type != "geo":
+            raise ValueError("'rule_type' must be 'geo'")
+
+        handlers = {
```

Comment:
Yeah great! This is now nicely compartmentalised for geodata rules only :) I like it.